docs/DeveloperGuide.md

```diff
@@ -1910,6 +1912,144 @@ testers are expected to do more *exploratory* testing.
        Expected: Similar to previous.
 
 
+### Session
+
+#### Creating a session
+
+1. Creating a session with a single student with _valid parameters_
+  1. Prerequisites: Student name must exist in the address book and no session with the session number given exists in the session list.
+  1. Test case: `createsession s/2 n/Alex Yeoh`<br>
+     Expected: A session with session number 2, containing the student Alex Yeoh, is created and stored in the session list.
+
+1. Creating a session with multiple students with _valid parameters_
+  1. Prerequisites: All student names must exist in the address book and no session with the session number given exists in the session list.
+  1. Test case: `createsession s/2 n/Alex Yeoh n/Bernice Yu`<br>
+     Expected: A session with session number 2, containing the students Alex Yeoh and Bernice Yu, is created and stored in the session list.
+
+1. Creating a session with a single student with _invalid parameters_
+  1. Test case: `createsession s/2 n/Alex Yeoh` `createsession s/2 n/Alex Yeoh`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No new session is created on the second command. <br>
+     Reason: There cannot exist two sessions with the same session number in the session list.
+  1. Test case: `createsession s/2 n/fakenamethatdoesnotexist`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No new session is created. <br>
+     Reason: Student name must exist in the address book.
+
+1. Creating a session with _mix of valid and invalid students_
+  1. Test case: `createsession s/2 n/Alex Yeoh n/fakenamethatdoesnotexist`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No new session is created. <br>
+     Reason: All student names must exist in the address book.
+
+1. Creating a session with _missing parameters_
+  1. Test case: `createsession s/2`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No new session is created. <br>
+     Reason: Student name argument is missing.
+  1. Test case: `createsession n/Alex Yeoh`<br>
+     Expected: F.A.K.E.J.A.R.V.I.S. displays an error. No new session is created. <br>
+     Reason: Session number argument is missing.
+
+
+#### Updating a session remark
+
+1. Updating a session remark with _valid parameters_
+  1. Prerequisites: Session must exist in the session list.
+  1. Test case: `updatesessionremark s/2 r/Update the remark to this text`<br>
+     Expected: The session with session number 2 has its remark field updated to "Update the remark to this text".
+
+1. Updating a session remark with _missing parameters_
```

Comment:
might want to standardise, parameters/arguments.

Reply:
alright, standardise with the rest to make it parameters